Assets/CarriageManager.cs

```diff
@@ -0,0 +1,143 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CarriageManager : MonoBehaviour
+{
+    public List<GameObject> carriagePrefabs;
+
+    private List<GameObject> carriagesToSpawn;
+    [SerializeField] private List<GameObject> spawnedCarriages;
+    [SerializeField] private CarriageData currentCarriage, oldCarriage;
+
+    //SPAWNING
+    [SerializeField] private Transform firstSpawn;
+    [SerializeField] private Transform nextSpawnPos,prevSpawnPos;
+
+    //MOVING
+    private void Awake()
+    {
+    }
+    private void Start()
+    {
+        InitialiseCarriages();
+    }
+
+    private void InitialiseCarriages()
+    {
+        //ALGORITHM CAN BE ALTERED TO BE RANDOMISED ON EVERY NEW PLAYTHROUGH OR KEPT AS IS
```

Comment:
you dont have to capitalise everything lol, we can all see comments perfectly fine when they're written normally 😄